packages/frontend/src/hooks/useChartConfig.tsx

```diff
@@ -106,7 +111,7 @@ export const useChartConfig = (queryResults: UseQueryResult<ApiQueryResults, Api
             setXDimension,
             setGroupDimension,
             toggleYMetric,
-            seriesLayout: {xDimension: seriesLayout.xDimension, yMetrics: seriesLayout.yMetrics, groupDimension: seriesLayout.groupDimension},  // verbose to help typescript
+            seriesLayout,
```

Comment:
I can't remember why this was needed, did typing work for you without it?

Reply:
The new function `isValidSeriesLayout` has 2 purposes:
- validates data
- confirms the input type ( notice the function return typing: `: seriesLayout is ValidSeriesLayout` )

```const isValidSeriesLayout = (seriesLayout: SeriesLayout): seriesLayout is ValidSeriesLayout => {
    return !!seriesLayout.xDimension && !!seriesLayout.yMetrics && seriesLayout.yMetrics.length > 0;
}```

So if the function returns true Typescript knows that inside the `if` block seriesLayout has type `ValidSeriesLayout` so we can remove that hack.